src/libraries/System.Net.Sockets/tests/FunctionalTests/ArgumentValidationTests.cs

```diff
@@ -365,6 +381,17 @@ public void Select_LargeList_Throws_ArgumentOutOfRange()
             Assert.Throws<ArgumentOutOfRangeException>(() => Socket.Select(null, null, largeList, -1));
         }
 
+        [Fact]
+        public void Select_LargeList_Throws_ArgumentOutOfRange_TimeSpan()
+        {
+            TimeSpan infinity = TimeSpan.FromMilliseconds(-1);
```

Comment:
this is not infinity. it is a timespan of negative 1 millisecond. TimeSpan cannot represent infinity. You can use TimeSpan.MaxValue if that applies.

Reply:
-1 milliseconds is used to mean "no timeout" or "infinite". Timeout.InfiniteTimeSpan returns such a timespan.

Reply:
Ugh -- I see it does. I assume that ultimately originates from windows INFINITE = 0xFFFFFFFF and some of the threading API wrapped those API, but accepting a signed integer. But TimeSpan's can be legitimately negative or positive, right? So in many cases -1 does not mean infinite. That seems unfortunate.

In that case then I suggest replacing `infinity` here with `Timeout.InfiniteTimeSpan` for clarity.